### PR TITLE
Changes on  #128 Spike: Setup SurrealDB test

### PR DESCRIPTION
**Pull request by muler-hussel**
I added a SurrealDbIntegrationTest, making some changes based on the guidance document provided by Testcontainers. In this project, subclasses of SurrealDbIntegrationTest would always initialize connection and driver with glabal configuration DatabaseConfig. If we follow the Testcontainers' guidance to manually initialize  connection and driver, a connection timeout may occur.

I added some changes to the date format problem. If we use timezone, then the startDate and endDate should be zonedDateTime. To deserialize and serialize zonedDateTime with Jackson, we should use DateTimeFormatter, while it seems settings in yml can only affect type Date.
DateTimeFormatter can only deserialize "Thu, May", one uppercase and two lowercase.
I wrote a global configuration, so the default deserialize and serialize that turn type Year into Integer might be overriden, so I changed Integer like 2024 in Json into String "2024".

### CODE REVIEW

**Comment by muler-hussel** [comment]
I added some changes to the date format problem. If we use timezone, then the startDate and endDate should be zonedDateTime. To deserialize and serialize zonedDateTime with Jackson, we should use DateTimeFormatter, while it seems settings in yml can only affect type Date.
DateTimeFormatter can only deserialize "Thu, May", one uppercase and two lowercase.
I wrote a global configuration, so the default deserialize and serialize that turn type Year into Integer might be overriden, so I changed Integer like 2024 in Json into String "2024".

File: src/main/java/com/wcc/platform/domain/platform/Event.java [hunk 1]
```diff
@@ -30,8 +31,8 @@ public class Event {
   @NotBlank private String title;
   @NotBlank private String description;
   @NotNull private EventType eventType;
-  @NotNull private String startDate; // TODO convert to Date
-  @NotNull private String endDate; // TODO convert to Date
+  @NotNull private ZonedDateTime startDate; // TODO convert to Date
```
Comment (dricazenck) [inline src/main/java/com/wcc/platform/domain/platform/Event.java:34]: you can remove the TODO now :)

File: src/main/java/com/wcc/platform/domain/platform/Event.java [hunk 2]
```diff
@@ -30,8 +31,8 @@ public class Event {
...
+  @NotNull private ZonedDateTime endDate; // TODO convert to Date
```
Comment (dricazenck) [inline src/main/java/com/wcc/platform/domain/platform/Event.java:35]: Same here :)

File: src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java [hunk 3]
```diff
@@ -1,52 +1,92 @@
...
   @BeforeAll
-  static void setUp() {
-    SURREAL_DB_CONTAINER.start();
+  static void setUpContainer() {
+    surrealDbContainer.start();
+
+    // Initialize SyncSurrealDriver with container connection
+    String host = surrealDbContainer.getHost();
+    Integer port = surrealDbContainer.getFirstMappedPort();
+    // String url = "ws://" + host + ":" + port + "/rpc";
```
Comment (dricazenck) [inline src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java:41]: you can remove this comment as well.

File: src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java [hunk 4]
```diff
@@ -1,52 +1,92 @@
...
+
+    connection = new SurrealWebSocketConnection(host, port, false);
```
Comment (dricazenck) [inline src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java:43]: I think here you should be using the Bean that initialize the connection, what do you think?

File: src/main/java/com/wcc/platform/deserializers/ZonedDateTimeDeserializer.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package com.wcc.platform.deserializers;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class ZonedDateTimeDeserializer extends JsonDeserializer<ZonedDateTime> {
```
Comment (dricazenck) [inline src/main/java/com/wcc/platform/deserializers/ZonedDateTimeDeserializer.java:10]: Please also add javadoc here

File: src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java [hunk 6]
```diff
@@ -1,52 +1,92 @@
...
-    registry.add("surrealdb.host", () -> host);
-    registry.add("surrealdb.port", port::toString);
-    registry.add("surrealdb.username", () -> "root");
-    registry.add("surrealdb.password", () -> "root");
-    registry.add("surrealdb.namespace", () -> "test_namespace");
-    registry.add("surrealdb.database", () -> "test_db");
-  }
+          .withCommand("start", "--log", "debug", "--user", "root", "--pass", "password");
+
+  static final String TABLE = "page";
```
Comment (dricazenck) [inline src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java:29]: this can be private right?

**Comment by dricazenck** [comment]
> I added some changes to the date format problem. If we use timezone, then the startDate and endDate should be zonedDateTime. To deserialize and serialize zonedDateTime with Jackson, we should use DateTimeFormatter, while it seems settings in yml can only affect type Date. DateTimeFormatter can only deserialize "Thu, May", one uppercase and two lowercase. I wrote a global configuration, so the default deserialize and serialize that turn type Year into Integer might be overriden, so I changed Integer like 2024 in Json into String "2024".

This information should be in the PR description, could you move there please?

**Comment by dricazenck** [comment]
Also please have a look in the build which is failing to run the [tests](https://github.com/Women-Coding-Community/wcc-backend/actions/runs/12403704625/job/34649311185?pr=166): 
image

File: src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java (hunk 4 above)
Comment (muler-hussel) [reply src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java:45]: Using the Bean may make connecting take more time, and cause initializationError in Build and test as seen in the picture you showed me. To solve the errors, I just changed test timeout in build.gradle to 5 minutes. If we use the Bean, errors would still occur.

File: src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java [hunk 7]
```diff
@@ -1,52 +1,94 @@
...
+    FooterPage page1 =
+        new FooterPage(
+            "id", "footer_title", "footer_subtitle", "footer_description", networks, link);
+
+    FooterPage page2 =
+        new FooterPage(
+            "id1", "footer1_title", "footer1_subtitle", "footer1_description", networks, link);
+
+    // Act
+    repository.save(page1);
```
Comment (dricazenck) [inline src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java:83]:
```suggestion
    repository.create(page1);
```


I think that was some conflicts after I merged my changes, could you please have a look, probably change save to create, if you run locally I think you will see the exception quickly :) 

But besides that everything looks great! Amazing work

File: src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java [hunk 8]
```diff
@@ -1,52 +1,94 @@
...
+    repository.save(page2);
```
Comment (dricazenck) [inline src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java:84]: After my changes also, you cannot create 2 footer pages, so I think this won't work anymore.

File: src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java [hunk 9]
```diff
@@ -1,52 +1,94 @@
...
+    Collection<FooterPage> pages = repository.findAll();
+
+    // Assert
+    assertNotNull(pages);
+    assertEquals(2, pages.size());
+
+    assertTrue(pages.stream().anyMatch(page -> page.id().equals(TABLE + ":" + page1.id())));
```
Comment (dricazenck) [inline src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java:91]:
```suggestion
    assertEquals(PageType.FOOTER.getPageId(), pages.first().id());
```

File: src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java [hunk 10]
```diff
@@ -1,52 +1,94 @@
...
+            "id", "footer_title", "footer_subtitle", "footer_description", networks, link);
+
+    FooterPage page2 =
+        new FooterPage(
+            "id1", "footer1_title", "footer1_subtitle", "footer1_description", networks, link);
+
+    // Act
+    repository.save(page1);
+    repository.save(page2);
+    Collection<FooterPage> pages = repository.findAll();
```
Comment (dricazenck) [inline src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java:85]:
```suggestion
var page = repository.findById(PageType.FOOTER.getPageId());
```

**Comment by dricazenck** [comment]
Please ping me on Slack when you are done the changes and I ca quickly review and we can merge :)

**Review by dricazenck — APPROVED** [review]

File: src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java [hunk 11]
```diff
@@ -1,52 +1,97 @@
...
-    SURREAL_DB_CONTAINER.start();
+  static void setUpContainer() {
+    surrealDbContainer.start();
+
+    // Initialize SyncSurrealDriver with container connection
+    String host = surrealDbContainer.getHost();
+    Integer port = surrealDbContainer.getFirstMappedPort();
+
+    connection = new SurrealWebSocketConnection(host, port, false);
+    connection.connect(1200); // timeout second
```
Comment (dricazenck) [inline src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java:44]:
```suggestion

```

**Pull request merged**